pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -433,23 +559,42 @@ func (m *manager) canResizePod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, stri
 		msg = "Node didn't have enough capacity: " + msg
 		klog.V(3).InfoS(msg, "pod", klog.KObj(pod))
 		return false, v1.PodReasonInfeasible, msg
+	}
 
+	for i := range allocatedPods {
+		for j, c := range allocatedPods[i].Status.ContainerStatuses {
+			actuatedResources, exists := m.GetActuatedResources(allocatedPods[i].UID, c.Name)
+			if exists {
+				// Overwrite the actual resources in the status with the actuated resources.
+				// This lets us reuse the existing scheduler libraries without having to wait
+				// for the actual resources in the status to be updated.
+				allocatedPods[i].Status.ContainerStatuses[j].Resources = &actuatedResources
+			}
+		}
 	}
 
 	if ok, failReason, failMessage := m.canAdmitPod(allocatedPods, pod); !ok {
-		// Log reason and return. Let the next sync iteration retry the resize
+		// Log reason and return.
 		klog.V(3).InfoS("Resize cannot be accommodated", "pod", klog.KObj(pod), "reason", failReason, "message", failMessage)
 		return false, v1.PodReasonDeferred, failMessage
 	}
 
 	return true, "", ""
 }
 
-// IsPodResizingInProgress checks whether the actuated resizable resources differ from the allocated resources
-// for any running containers. Specifically, the following differences are ignored:
-// - Non-resizable containers: non-restartable init containers, ephemeral containers
-// - Non-resizable resources: only CPU & memory are resizable
-// - Non-running containers: they will be sized correctly when (re)started
+func (m *manager) CheckPodResizeInProgress(allocatedPod *v1.Pod, podStatus *kubecontainer.PodStatus) {
```

Comment:
Let's leave this here for now, but maybe we should move all the actuated resources & in-progress logic out of allocation manager in a follow-up PR. It can almost all go to the runtime. That would also break the circular dependency with the kuberuntime. WDYT?

I need to think about this more.